cmd/kubeadm/app/preflight/checks.go

```diff
@@ -803,7 +803,9 @@ func getEtcdVersionResponse(client *http.Client, url string, target interface{})
 				loopCount--
 				return false, err
 			}
-			defer r.Body.Close()
+			if r != nil {
```

Comment:
i don't think this can happen.
for `err != nil` we are already returning.

Reply:
yes, it looks never can happen, but it depends on  client.Get() always ensure err will not be nil if response is nil. do you prefer to add hint to ignore this staticcheck error? or keep the status quo ? thanks.

Reply:
> yes, it looks never can happen, but it depends on client.Get() always ensure err will not be nil if response is nil

this is pretty much the case in the Go source code.

> do you prefer to add hint to ignore this staticcheck error?

yes, i'd prefer if this is marked as a false-positive or ideally the static check tool should be updated / fixed.

Reply:
updated.thanks.